### PR TITLE
Fix homepage content switcher size

### LINKED ISSUE #646: Align content switcher inside search bar with Figma mockups
## Description

The content switcher inside the homepage search bar does not match the Figma mockups.

## Expectation

It should match.

## Additional context

This issue was branched from https://github.com/WordPress/openverse-frontend/pull/593#issuecomment-1016501636 and pertains to the switcher added in #613.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #646 by @dhruvkb

## Description

This PR makes the content switcher in the desktop version of the homepage taller to match the Figma mockups. It also changes the end padding for desktop version to make it balanced, considering that the arrow down anchor also adds a tiny padding on desktop.

## Testing Instructions

Open the homepage with a screen width larger than `md`. The search bar should have a content switcher inside that is 48px high. 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by sarayourfriend — APPROVED** [review]
Looks great to me!

File: src/components/VContentSwitcher/VContentSwitcherButton.vue [hunk 1]
```diff
@@ -52,13 +57,28 @@ export default {
...
+        return 'h-12 px-2'
+      } else if (isIconButton.value) {
+        return 'w-10 h-10'
+      } else if (isMinScreenMd.value) {
+        // When there is a caret down icon, paddings are balanced
+        return 'px-2'
+      } else {
+        // Without caret down icon, paddings need to be adjusted
+        return 'ps-2 pe-3'
+      }
```
Comment (krysal) [inline src/components/VContentSwitcher/VContentSwitcherButton.vue:71]: Seems that the last two conditions can be combined and simplify the code.
```suggestion
      } else {
        /**
          When there is a caret down icon (on 'md' screens), paddings are balanced, 
          without it, paddings need to be adjusted.
          */
        return 'ps-2 pe-3 md:px-2'
      }
```

File: src/components/VContentSwitcher/VContentSwitcherPopover.vue [hunk 2]
```diff
@@ -9,6 +9,11 @@
       <VContentSwitcherButton
         :a11y-props="a11yProps"
         :active-item="activeItem"
+        :class="{
+          '!border-tx group-hover:!border-dark-charcoal-06':
```
Comment (krysal) [inline src/components/VContentSwitcher/VContentSwitcherPopover.vue:13]: Great idea to use `!important`! I didn't know Tailwind can do it before seeing this. The [documentation](https://tailwindcss.com/docs/configuration#important) says it needs to be configured and I don't see that in our `tailwind.config.js` file so maybe with JIT mode it is not necessary?

Although the border is almost imperceptible, shouldn't that be the same value as the border of the input field? Sure @panchovm should know better.
```suggestion
          '!border-tx group-hover:!border-dark-charcoal-20':
```

**Review by krysal — COMMENTED** [review]
It's looking greaaaat. I just have a question about the border and probably @panchovm would want to review it too.

File: src/components/VContentSwitcher/VContentSwitcherPopover.vue (hunk 2 above)
Comment (panchovm) [reply src/components/VContentSwitcher/VContentSwitcherPopover.vue:13]: You are right @krysal, in `hover` the border should be `dark-charcoal-20`.

**Comment by panchovm** [comment]
The content switcher is and behaves like the original one, I was confused when reading the description "the content switcher in the desktop version of the homepage taller to match the Figma mockups". But the component looks like the mockups. It might be a little misunderstanding. 

However, I noticed there is a border change when going over the active button.



I would check if in all component states you set the `2px` radius corner.

Despite the above, it looks great ✨

File: src/components/VContentSwitcher/VContentSwitcherPopover.vue (hunk 2 above)
Comment (obulat) [reply src/components/VContentSwitcher/VContentSwitcherPopover.vue:13]: And I was looking on the border trying to see why I can't really see it, @krysal ! Thank you so much for spotting this!

**Comment by obulat** [comment]
> The content switcher is and behaves like the original one, I was confused when reading the description "the content switcher in the desktop version of the homepage taller to match the Figma mockups". But the component looks like the mockups. It might be a little misunderstanding.

I've fixed the border a little bit, and here is the comparison of the current implementation of switcher on the main branch (button always has a border, and it's 42px tall instead of 48px, and when pressed, its border is incorrect):
_Please disregard the search button label changes_

content-switcher-current

With the changes in this PR:
- button height is 48px
- button doesn't have a border when resting
content-switcher-new

Sorry for not being clear from the start, @panchovm

**Review by zackkrida — APPROVED** [review]
Looks nice!

**Review by krysal — APPROVED** [review]
Nice!

**Pull request merged**